File: ga/gpt3_5_result_2024_6/result/1968B_gpt/18_1968B.c
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let mut input = stdin.lock();
    
    let mut input_line = String::new();
    input.read_line(&mut input_line).unwrap();
    let t: i32 = input_line.trim().parse().unwrap();

    for _ in 0..t {
        input_line.clear();
        input.read_line(&mut input_line).unwrap();
        let mut parts = input_line.trim().split_whitespace();
        let n: usize = parts.next().unwrap().parse().unwrap();
        let m: usize = parts.next().unwrap().parse().unwrap();

        input_line.clear();
        input.read_line(&mut input_line).unwrap();
        let sn = input_line.trim();

        input_line.clear();
        input.read_line(&mut input_line).unwrap();
        let sm = input_line.trim();
        
        let mut i = 0;
        let mut j = 0;
        while i < n && j < m {
            if sn.chars().nth(i) == sm.chars().nth(j) {
                i += 1;
            }
            j += 1;
        }

        println!("{}", i);
    }
}

File: ga/gpt3_5_result_2024_6/result/1968B_cpp_gpt/18_1968B.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let mut iterator = stdin.lock().lines().map(Result::unwrap);

    let t: usize = iterator.next().unwrap().parse().unwrap();

    for _ in 0..t {
        let input: Vec<i32> = iterator.next().unwrap().split_whitespace()
            .map(|x| x.parse().unwrap()).collect();
        let x = input[0] as usize;
        let k = input[1] as usize;

        let a: Vec<char> = iterator.next().unwrap().chars().collect();
        let b: Vec<char> = iterator.next().unwrap().chars().collect();
        
        let mut ans = 0;
        for i in 0..k {
            if i < x && a[i] == b[i] {
                ans += 1;
            }
        }
        println!("{}", ans);
    }
}